pkg/webhooks/server.go

```diff
@@ -158,12 +254,34 @@ func protect(inner handlers.AdmissionHandler) handlers.AdmissionHandler {
 	}
 }
 
+func logAdmission(inner handlers.AdmissionHandler) handlers.AdmissionHandler {
+	return func(logger logr.Logger, request *admissionv1.AdmissionRequest, startTime time.Time) *admissionv1.AdmissionResponse {
+		resp := inner(logger, request, startTime)
+		if dumpPayload {
+			if dumpOperation(string(request.Operation)) && dumpKind(request.Kind.Kind) {
+				reqPayload, err := newAdmissionRequestPayload(request)
+				if err != nil {
+					logger.Error(err, "Failed to extract resources")
+					return admissionutils.ResponseFailure(err.Error())
+				}
+				if strings.EqualFold(request.Kind.Kind, "Secret") {
+					redactSecretKind(reqPayload)
+					go logger.Info("logging admission review payload", "AdmissionRequest", reqPayload, "AdmissionReponse", resp)
+				} else {
+					go logger.Info("logging admission review payload", "AdmissionRequest", reqPayload, "AdmissionResponse", resp)
+				}
+			}
+		}
+		return resp
+	}
+}
+
 func filter(configuration config.Configuration, inner handlers.AdmissionHandler) handlers.AdmissionHandler {
 	return handlers.Filter(configuration, inner)
 }
 
 func admission(logger logr.Logger, monitor *webhookconfig.Monitor, inner handlers.AdmissionHandler) http.HandlerFunc {
-	return handlers.Monitor(monitor, handlers.Admission(logger, protect(inner)))
+	return handlers.Monitor(monitor, handlers.Admission(logger, protect(logAdmission(inner))))
```

Comment:
You need to inject a logger here.

Reply:
alright

Reply:
@eddycharly do you mean the logger that is being passed from `handlers.Admission` should not be used in `logAdmission`? If yes, should the separate logger instance for logAdmission be a logger.WithName("debugmode") or not?

Reply:
You're right, it's ok with the current implementation.